xlml/utils/tpu.py

```diff
@@ -140,6 +143,9 @@ def create_queued_resource_request(tpu_name: str, ssh_keys: ssh.SshKeys) -> str:
     logging.info('Create QR response: {}'.format(response))
     # TODO(wcromar): do anything about failures
 
+    log_url = f'https://console.cloud.google.com/logs/query;query=resource.type%3D%22gce_instance%22%20AND%0Alog_id%2528%22{tpu_name}%22%2529;?project={gcp.project_name}'
```

Comment:
For formatting complicated HTTP query strings like this, you can use `urlencode`: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode

That lets you build a more natural multi-line string in Python for the Cloud logging query before it gets mangled.

Reply:
Sounds good. Will take a look